src/library/scala/collection/SortedMap.scala

```diff
@@ -223,4 +223,4 @@ object SortedMapOps {
 }
 
 @SerialVersionUID(3L)
-object SortedMap extends SortedMapFactory.Delegate[SortedMap](TreeMap)
+object SortedMap extends SortedMapFactory.Delegate[SortedMap](immutable.SortedMap)
```

Comment:
what's the reason for this change?

Reply:
We can return the argument collection from `from` if and only if the argument is an immutable sorted map. That means that the 'from' logic is identical to the immutable sorted map object, so we should either copy paste that 'from' implementation here, or just delegate to that one

Reply:
the other ones all seem to delegate to the concrete collection

Reply:
`collection.{Seq, LinearSeq, IndexedSeq}` also delegate to `immutable.{Seq, LinearSeq, IndexedSeq}`, though when you wrote this comment, `collection.LinearSeq` did delegate directly to `immutable.List`.